api/src/app/modules/project/project.service.ts

```diff
@@ -784,6 +784,105 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
     return true;
   }
 
+  /**
+   * Counts the number of projects where commentingOpenDate falls between startDate and endDate. 
+   * Used by analytics dashboard module.
+   * 
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to the count of projects matching the criteria
+   */
+  async getProjectCountByDate(startDate: string, endDate: string): Promise<number> {
+    return await this.repository
+      .createQueryBuilder()
+      .where('commenting_open_date >= :startDate', { startDate })
+      .andWhere('commenting_open_date <= :endDate', { endDate })
+      .andWhere('workflow_state_code != :workflowStateCode', { 
+        workflowStateCode: WorkflowStateEnum.INITIAL 
+      })
+      .getCount();
+  }
+
+
+  /**
+   * Retrieves the count of FOM projects grouped by district.
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to an array of FOMCountByDistrictDto
+   */
+  async getProjectCountByDistrict(startDate: string, endDate: string): Promise<ProjectCountByDistrictResponse[]> {
+    return await this.repository
+      .createQueryBuilder('p')
+      .select('p.district_id', 'districtId')
+      .where('p.commenting_open_date >= :startDate', { startDate })
+      .andWhere('p.commenting_open_date <= :endDate', { endDate })
+      .andWhere('p.workflow_state_code != :workflowStateCode', { 
+        workflowStateCode: WorkflowStateEnum.INITIAL 
+      })
+      .innerJoin('p.district', 'd')
+      .addSelect('COUNT(p.project_id)', 'projectCount')
+      .addSelect('d.name', 'districtName')
+      .groupBy('p.district_id')
+      .addGroupBy('d.name')
+      .orderBy('"projectCount"', 'DESC')
+      .getRawMany();
+  }
+
+  /**
+   * Returns the total number of unique forest clients that have submitted FOMs.
+   * Excludes records where forest_client_number is null. Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to the count of distinct forest client numbers
+   */
+  async getUniqueForestClientCount(startDate: string, endDate: string): Promise<number> {
+    return await this.repository
+      .createQueryBuilder()
+      .select('COUNT(DISTINCT forest_client_number)', 'count')
+      .where('forest_client_number IS NOT NULL')
+      .andWhere('commenting_open_date >= :startDate', { startDate })
+      .andWhere('commenting_open_date <= :endDate', { endDate })
+      .andWhere('workflow_state_code != :workflowStateCode', { 
```

Comment:
So when we count the number of unique forest client numbers that submitted FOM, not sure if we still require `commenting_open_date` within the selected date range, and excluding "**INITIAL**" FOMs.

Reply:
Date logic yes, initial FOMS would already be excluded if you are looking for a different status corresponding to finalized (submitted to ministry).